include/seqan/blast/blast_report_out.h

```diff
@@ -0,0 +1,893 @@
+ *
+ * The reference Blast implementation used for developing the SeqAn support is NCBI Blast+ 2.2.26. In contrast to the
+ * tabular format there is no support for writing legacy files (without the +).
+ *
+ * SeqAn also supports reading and writing the tabular blast output format, see @link BlastTabular @endlink.
+ *
+ * @section Example of high-level file writing
+ *
+ * You can use this tag's interface and specify the stream, and context:
+ * <li> BlastReport#@link BlastReport#writeHeader @endlink </li>
+ * <li> BlastReport#@link BlastReport#writeRecord @endlink up to n times</li>
+ * <li> BlastReport#@link BlastReport#writeFooter @endlink </li>
+ *
+ * Or you can use the FormattedFile interface which has the same functions, but only requires one parameter:
+ * <li> BlastReportOut#@link BlastReportOut#writeHeader @endlink </li>
+ * <li> BlastReportOut#@link BlastReportOut#writeRecord @endlink up to n times</li>
+ * <li> BlastReportOut#@link BlastReportOut#writeFooter @endlink </li>
+ *
+ * See @link BlastReportOut @endlink for a full code example.
+ */
+
+struct BlastReport_;
+typedef Tag<BlastReport_> BlastReport;
+
+// ----------------------------------------------------------------------------
+// Tag BlastReportOut
+// ----------------------------------------------------------------------------
+
+/*!
+ * @class BlastReportOut
+ * @signature template <typename TBlastIOContext = BlastIOContext<>>
+ * using BlastReportOut = FormattedFile<BlastReport, Output, TBlastIOContext>;
+ * @extends FormattedFileOut
+ * @headerfile <seqan/blast.h>
+ * @brief FormattedFileOut abstraction for @link BlastReport @endlink
+ *
+ * This is a @link FormattedFile @endlink specialization for writing @link BlastReport @endlink formats. For details
+ * on how to influence the writing of files, see @link BlastIOContext @endlink.
+ * Please note that you have to specify the type of the context as a template parameter to BlastReportOut, see the
+ * example below.
+ *
+ * @section Example
+ *
+ * The following short program creates the pairwise alignments between three query sequences and two database sequences,
+ * it computes the e-values, sorts the matches and prints all results that score above a threshold. <i>The same example
+ * is used for @link BlastTabularOut @endlink and @link BlastReportOut @endlink, you only need to change one line.</i>
+ *
+ * @include demos/blast/blast_out_example.cpp
+ *
+ * The file generated in /tmp/output.blast looks like this:
+ *
+ * @include demos/blast/blast_out_example.report
+ *
+ * @see BlastRecord
+ */
+
+template <typename TBlastIOContext = BlastIOContext<> >
+using BlastReportOut = FormattedFile<BlastReport, Output, TBlastIOContext>;
+
+// ----------------------------------------------------------------------------
+// Class MagicHeader
+// ----------------------------------------------------------------------------
+
+template <typename T>
+struct MagicHeader<BlastReport, T> :
+    public MagicHeader<Nothing, T> {};
```

Comment:
> Really? Isn't it a string like BLAST*?

It is '> BLAST*' or '> TBLAST*' for tabular with header, it is not defined for tabular without header, it  is 'BLAST*' or 'TBLAST*' for report.

> FYI the public keyword is superfluous when you inherit a struct.

Ah, I didn't know, but it makes sense of course.